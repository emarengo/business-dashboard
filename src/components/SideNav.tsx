import React from 'react';
import { Link, withRouter } from 'react-router-dom';

import Styled from './styled';
import SvgIcons from './svgs';
import { ISideNav } from './types';

const CustomSideNav: React.FC<ISideNav> = ({
  home,
  location,
  isDriverPerformanceServiceEnabled
}) => {
  const isCurrentPage = (slug: string): boolean =>
    new RegExp(`^/?${slug}`).test(location.pathname) ||
    (slug === home && location.pathname === '/');

  return (
    <Styled.SideNav showLogo>
      <Link to="/shifts">
        <Styled.NavListItem isActive={isCurrentPage('shifts')}>
          <Styled.NavListItemIcon>
            <SvgIcons.Calendar />
          </Styled.NavListItemIcon>
        </Styled.NavListItem>
      </Link>
      <Link to="/vehicles">
        <Styled.NavListItem isActive={isCurrentPage('vehicles')}>
          <Styled.NavListItemIcon>
            <SvgIcons.Car />
          </Styled.NavListItemIcon>
        </Styled.NavListItem>
      </Link>
      {isDriverPerformanceServiceEnabled && (
        <Link to="/drivers">
          <Styled.NavListItem isActive={isCurrentPage('drivers')}>
            <Styled.NavListItemIcon>
              <SvgIcons.Wheel />
            </Styled.NavListItemIcon>
          </Styled.NavListItem>
        </Link>
      )}
      <Link to="/geofence/1">
        <Styled.NavListItem isActive={isCurrentPage('geofence')}>
          <Styled.NavListItemIcon>
            <SvgIcons.Map />
          </Styled.NavListItemIcon>
        </Styled.NavListItem>
      </Link>
      <Link to="/settings">
        <Styled.NavListItem isActive={isCurrentPage('settings')}>
          <Styled.NavListItemIcon>
            <SvgIcons.Gear />
          </Styled.NavListItemIcon>
        </Styled.NavListItem>
      </Link>
    </Styled.SideNav>
  );
};

export default withRouter(CustomSideNav);
