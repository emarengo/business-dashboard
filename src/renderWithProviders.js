import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { ThemeProvider } from 'emotion-theming';
import { BeatTheme } from '@thebeatapp/beat-ui';

const customRender = (node, ...options) => {
  return render(
    <Router>
      <ThemeProvider theme={BeatTheme}>{node}</ThemeProvider>
    </Router>,
    ...options
  );
};

export * from '@testing-library/react';
export { customRender as render };
export { customRender as rerender };
