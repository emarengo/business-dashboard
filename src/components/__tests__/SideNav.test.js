import React from 'react';
import { cleanup, render } from '../../renderWithProviders';
import SideNav from '../SideNav';

describe('SideNav component', () => {
  let renderResult;

  beforeEach(() => {
    renderResult = render(<SideNav />);
  });

  afterEach(cleanup);

  it('should match snapshot', () => {
    expect(renderResult).toMatchSnapshot();
  });
});
