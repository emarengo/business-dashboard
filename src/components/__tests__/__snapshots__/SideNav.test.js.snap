// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav component should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      class="beat-ui-portal"
    />
    <div>
      <nav
        class="beat-ui-sidenav css-i31tm3 e1hw2vfw0"
        data-testid="beat-sidenav"
        id=""
      >
        <div
          class="css-mkre44 e1hw2vfw1"
          data-testid="side-nav-logo"
        >
          <svg
            fill="#23d2aa"
            height="23"
            viewBox="0 0 19 28"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10.299 22.4H6.182c-.247 0-.412-.16-.412-.4v-5.4c0-.24.165-.4.412-.4H10.3c1.73 0 2.884 1.52 2.884 3.08 0 1.6-1.112 3.12-2.884 3.12m2.56-14.16c0 1.4-1.278 2.68-2.72 2.68H6.182c-.247 0-.412-.16-.412-.4V6c0-.24.165-.4.412-.4h3.957c1.442 0 2.72 1.24 2.72 2.64m3.71 5.68c-.248-.16-.248-.48 0-.72 1.483-1.44 2.102-3.28 2.102-5.2 0-4.36-3.504-8-8.532-8H2.746C1.229 0 0 1.193 0 2.664v22.672C0 26.807 1.23 28 2.746 28h7.393C15.621 28 19 24.6 19 19.28c0-1.96-.742-4.08-2.432-5.36"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <ul
          class="css-1ylu0bo e1hw2vfw2"
        >
          <a
            href="/shifts"
          >
            <li
              class="beat-ui-sidenav-item css-xqhjbi e1hw2vfw3"
              data-testid="nav-list-item"
            >
              <div
                class="css-16oynd e1hw2vfw5"
              >
                <svg
                  height="15"
                  viewBox="0 0 18 15"
                  width="18"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16.875 0C17.465 0 18 .562 18 1.213v12.604c0 .62-.534 1.183-1.125 1.183H1.125C.535 15 0 14.438 0 13.817V1.183C0 .473.534 0 1.125 0zm-.394 5.533H1.491v7.899h14.99v-7.9zM4.894 10.266c.225 0 .365.148.365.385v.799c0 .236-.14.384-.365.384h-.76c-.225 0-.393-.148-.365-.384v-.8c0-.236.14-.384.365-.384zm2.981 0c.225 0 .366.148.366.385v.799c0 .236-.141.384-.366.384h-.76c-.224 0-.365-.148-.365-.384v-.8c0-.236.14-.384.366-.384zm3.01 0c.224 0 .365.148.365.385v.799c0 .236-.14.384-.366.384h-.759c-.225 0-.366-.148-.366-.384v-.8c0-.236.141-.384.366-.384zm3.009 0c.225 0 .365.148.365.385v.799c0 .236-.14.384-.365.384h-.76c-.225 0-.393-.148-.365-.384v-.8c0-.236.14-.384.365-.384zm-9-3.136c.225 0 .365.148.365.385v.799c0 .236-.14.384-.365.384h-.76c-.225 0-.393-.177-.365-.384v-.8c0-.236.14-.384.365-.384zm2.981 0c.225 0 .366.148.366.385v.799c0 .236-.141.384-.366.384h-.76c-.224 0-.365-.177-.365-.384v-.8c0-.236.14-.384.366-.384zm3.01 0c.224 0 .365.148.365.385v.799c0 .236-.14.384-.366.384h-.759c-.225 0-.366-.177-.366-.384v-.8c0-.236.141-.384.366-.384zm3.009 0c.225 0 .365.148.365.385v.799c0 .236-.14.384-.365.384h-.76c-.225 0-.393-.177-.365-.384v-.8c0-.236.14-.384.365-.384zm2.587-5.532H1.491v2.366h14.99V1.598z"
                    fill="#242C39"
                  />
                </svg>
              </div>
            </li>
          </a>
          <a
            href="/vehicles"
          >
            <li
              class="beat-ui-sidenav-item css-xqhjbi e1hw2vfw3"
              data-testid="nav-list-item"
            >
              <div
                class="css-16oynd e1hw2vfw5"
              >
                <svg
                  height="14"
                  viewBox="0 0 19 14"
                  width="19"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    fill="none"
                    fill-rule="evenodd"
                  >
                    <g
                      fill="#242C39"
                      fill-rule="nonzero"
                    >
                      <g>
                        <path
                          d="M26.619 179l.366.005 6.005.022c.481-.025.964.012 1.435.112.728.204 1.325.723 1.625 1.413l1.244 3.244c.03.085.088.159.164.21.277.187.963.66 1.11.87.182.267.31.567.38.882.022.117.043 2.008.047 3.34v.347c0 .247.018.647.018.894 0 .456-.04.88-.349 1.25-.326.366-.804.562-1.296.53-.528 0-1.012-.01-1.428-.39-.417-.38-.48-.871-.48-1.399 0-.247-.201-.447-.45-.447h-9.999c-.248 0-.45.2-.45.447-.004.53-.063 1.01-.474 1.395-.365.342-.817.4-1.283.4-.466 0-.898-.031-1.276-.353-.353-.297-.482-.68-.52-1.102l-.008-.147.003-.191c.02-1.653.074-4.582.19-5.033l.013-.042c.168-.357.415-.672.723-.92.178-.124.364-.235.558-.334.095-.05.168-.135.205-.237l.51-1.363.329-.87c.204-.55.373-1.133.756-1.594.405-.483.975-.802 1.602-.894.364-.04.73-.053 1.096-.04zm10.296 6.05c-.143-.086-.319-.094-.47-.023l-1.685.796c-.079.037-.142.092-.19.156h-3.769l-.8.84-.8-.84h-3.803c-.047-.064-.11-.119-.19-.156l-1.685-.796-.093-.033c-.126-.032-.262-.013-.376.056-.255.152-.432.367-.5.637l-.042.176-.014.077c-.005.045-.005.084-.002.142.007.11.034.206.083.32.09.21.254.376.476.507.077.046.165.07.255.07h1.434l.085-.007c.084-.015.163-.05.23-.105l.251-.204.035-.034H28.8l1.2 1.238 1.2-1.238h3.422l.036.034.251.204c.09.073.2.112.315.112h1.434c.09 0 .178-.024.255-.07.222-.131.386-.298.476-.508.049-.113.076-.209.083-.32.005-.085.003-.13-.016-.218l-.042-.176c-.068-.27-.245-.485-.5-.637zm-3.092-4.55h-7.686c-.44 0-.828.287-.957.708l-.503 1.646c-.014.047-.021.096-.021.146 0 .276.223.5.5.5h9.648c.05 0 .1-.007.147-.022.264-.08.412-.36.332-.624l-.503-1.646c-.129-.42-.517-.708-.957-.708z"
                          transform="translate(-21 -179)"
                        />
                      </g>
                    </g>
                  </g>
                </svg>
              </div>
            </li>
          </a>
          <a
            href="/geofence/1"
          >
            <li
              class="beat-ui-sidenav-item css-xqhjbi e1hw2vfw3"
              data-testid="nav-list-item"
            >
              <div
                class="css-16oynd e1hw2vfw5"
              >
                <svg
                  height="16"
                  viewBox="0 0 20 16"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M18.847 3.91l-4.324-2.85c-.12-.08-.278-.08-.398 0L10 3.777 5.875 1.06c-.12-.079-.278-.079-.398 0L1.153 3.91c-.096.063-.153.168-.153.28v10.472c0 .126.072.241.188.3.116.058.255.05.363-.02l4.125-2.72 4.125 2.72c.12.079.278.079.398 0l4.125-2.72 4.125 2.72c.06.04.13.059.199.059.056 0 .112-.013.164-.04.116-.058.188-.173.188-.299V4.189c0-.112-.057-.217-.153-.28zm-.552 6.597l-1.549-3.065 1.55-.158v3.223h0zm0-3.563l-3.707.377V1.924l3.707 2.444v2.576h0zm-16.59-.2h3.707v3.221l-3.707-.493V6.744zm8.03-2.318v6.127L6.839 7.536 8.804 3.81l.932.615zm2.154-1.071l2.17.908v3.25l-3.795 3.125V4.426l1.625-1.071zm2.17.538l-1.827-.764 1.828-1.205v1.97h0zM6.585 7.27L5.94 6.6V1.924l2.57 1.694L6.585 7.27zM5.94 7.1l3.796 3.953v3.023L5.94 11.574V7.1zm4.324 3.984l3.796-3.125v3.615l-3.796 2.502v-2.992zm-4.852-9.16v4.481H1.705V4.368l3.707-2.444zm-3.707 7.89l3.707.494v1.266l-3.707 2.444V9.814zm12.883 1.76V7.662l1.786-.182 1.921 3.803v2.735l-3.707-2.444z"
                    fill="#242C39"
                    stroke="#242C39"
                    stroke-width=".6"
                  />
                </svg>
              </div>
            </li>
          </a>
          <a
            href="/settings"
          >
            <li
              class="beat-ui-sidenav-item css-xqhjbi e1hw2vfw3"
              data-testid="nav-list-item"
            >
              <div
                class="css-16oynd e1hw2vfw5"
              >
                <svg
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10.663 1c.669 0 1.212.544 1.212 1.212h0v1.345c0 .205.12.391.29.452.188.068.371.144.549.228.158.076.368.03.51-.112h0l.957-.958c.473-.471 1.241-.471 1.715 0h0l.937.938c.473.472.473 1.241 0 1.714h0l-.958.958c-.14.141-.186.35-.111.51.084.178.16.36.227.547.061.171.248.29.453.29h1.345c.668 0 1.212.544 1.211 1.213h0v1.326c0 .669-.544 1.212-1.212 1.212h0-1.345c-.205 0-.391.12-.453.29-.068.188-.144.37-.228.548-.075.16-.03.369.112.51h0l.958.958c.473.473.472 1.242 0 1.714h0l-.937.938c-.473.47-1.241.472-1.715 0h0l-.957-.958c-.14-.141-.35-.188-.51-.112-.178.085-.36.16-.547.227-.172.063-.291.249-.291.453h0v1.345c0 .668-.544 1.212-1.212 1.212h0-1.326c-.669 0-1.212-.543-1.212-1.212h0v-1.345c0-.205-.12-.391-.29-.453-.188-.068-.371-.144-.549-.228-.159-.075-.368-.029-.51.112h0l-.957.958c-.473.47-1.241.472-1.715 0h0l-.937-.938c-.472-.472-.472-1.24 0-1.714h0l.958-.958c.141-.14.187-.35.112-.51-.084-.177-.16-.36-.227-.546-.062-.172-.248-.291-.453-.291h0-1.345C1.544 11.875 1 11.33 1 10.663h0V9.337c0-.669.544-1.212 1.212-1.212h1.345c.205 0 .391-.12.452-.29.068-.188.144-.37.228-.548.075-.16.03-.369-.11-.51h0l-.96-.958c-.472-.473-.472-1.242 0-1.714h0l.937-.938c.473-.471 1.242-.471 1.715 0h0l.958.958c.14.141.35.187.51.112.178-.085.36-.16.547-.227.172-.063.291-.249.291-.453h0V2.212C8.125 1.544 8.67 1 9.337 1h0zm0 .75H9.337c-.255 0-.462.207-.462.462h0v1.345c0 .523-.315.99-.786 1.158-.164.06-.324.126-.481.2-.448.212-.994.106-1.36-.26h0l-.959-.957c-.18-.18-.474-.18-.654 0h0l-.937.937c-.18.18-.18.473 0 .653h0l.958.958c.366.365.47.911.26 1.36-.075.157-.141.318-.201.483-.169.47-.635.786-1.158.786h0-1.345c-.255 0-.462.207-.462.462h0v1.326c0 .255.207.462.462.462h1.345c.523 0 .99.315 1.159.786.059.165.126.325.2.483.21.447.106.994-.26 1.36h0l-.958.956c-.18.18-.18.474 0 .654h0l.937.938c.182.18.475.18.654 0h0l.958-.958c.241-.24.558-.368.875-.368.166 0 .331.036.485.108.158.075.318.141.483.201.47.17.785.635.785 1.158h0v1.345c0 .255.207.462.462.462h1.326c.255 0 .462-.207.462-.462h0v-1.345c0-.523.316-.989.786-1.158.165-.06.325-.126.482-.2.448-.21.995-.106 1.36.26h0l.958.957c.182.18.475.18.654 0h0l.938-.937c.18-.18.18-.474 0-.654h0l-.958-.958c-.366-.365-.47-.911-.26-1.36.075-.157.142-.317.2-.482.17-.47.635-.786 1.159-.786h1.345c.255 0 .462-.207.462-.462h0V9.337c0-.255-.207-.462-.462-.462h0-1.345c-.524 0-.99-.316-1.159-.787-.06-.164-.126-.325-.2-.482-.21-.448-.106-.995.26-1.36h0l.958-.958c.18-.18.18-.473 0-.653h0l-.938-.937c-.18-.18-.474-.18-.654 0h0l-.957.957c-.366.367-.913.472-1.36.26-.158-.074-.318-.14-.483-.2-.47-.17-.786-.635-.786-1.158h0V2.212c0-.255-.207-.462-.462-.462h0zM10 7.375c1.448 0 2.625 1.177 2.625 2.625 0 1.447-1.177 2.625-2.625 2.625-1.447 0-2.625-1.178-2.625-2.625 0-1.448 1.178-2.625 2.625-2.625zm0 .75c-1.033 0-1.875.84-1.875 1.875 0 1.034.842 1.875 1.875 1.875 1.034 0 1.875-.841 1.875-1.875 0-1.035-.841-1.875-1.875-1.875z"
                    fill="#242C39"
                    stroke="#242C39"
                    stroke-width=".6"
                  />
                </svg>
              </div>
            </li>
          </a>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="beat-ui-sidenav css-i31tm3 e1hw2vfw0"
      data-testid="beat-sidenav"
      id=""
    >
      <div
        class="css-mkre44 e1hw2vfw1"
        data-testid="side-nav-logo"
      >
        <svg
          fill="#23d2aa"
          height="23"
          viewBox="0 0 19 28"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.299 22.4H6.182c-.247 0-.412-.16-.412-.4v-5.4c0-.24.165-.4.412-.4H10.3c1.73 0 2.884 1.52 2.884 3.08 0 1.6-1.112 3.12-2.884 3.12m2.56-14.16c0 1.4-1.278 2.68-2.72 2.68H6.182c-.247 0-.412-.16-.412-.4V6c0-.24.165-.4.412-.4h3.957c1.442 0 2.72 1.24 2.72 2.64m3.71 5.68c-.248-.16-.248-.48 0-.72 1.483-1.44 2.102-3.28 2.102-5.2 0-4.36-3.504-8-8.532-8H2.746C1.229 0 0 1.193 0 2.664v22.672C0 26.807 1.23 28 2.746 28h7.393C15.621 28 19 24.6 19 19.28c0-1.96-.742-4.08-2.432-5.36"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <ul
        class="css-1ylu0bo e1hw2vfw2"
      >
        <a
          href="/shifts"
        >
          <li
            class="beat-ui-sidenav-item css-xqhjbi e1hw2vfw3"
            data-testid="nav-list-item"
          >
            <div
              class="css-16oynd e1hw2vfw5"
            >
              <svg
                height="15"
                viewBox="0 0 18 15"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16.875 0C17.465 0 18 .562 18 1.213v12.604c0 .62-.534 1.183-1.125 1.183H1.125C.535 15 0 14.438 0 13.817V1.183C0 .473.534 0 1.125 0zm-.394 5.533H1.491v7.899h14.99v-7.9zM4.894 10.266c.225 0 .365.148.365.385v.799c0 .236-.14.384-.365.384h-.76c-.225 0-.393-.148-.365-.384v-.8c0-.236.14-.384.365-.384zm2.981 0c.225 0 .366.148.366.385v.799c0 .236-.141.384-.366.384h-.76c-.224 0-.365-.148-.365-.384v-.8c0-.236.14-.384.366-.384zm3.01 0c.224 0 .365.148.365.385v.799c0 .236-.14.384-.366.384h-.759c-.225 0-.366-.148-.366-.384v-.8c0-.236.141-.384.366-.384zm3.009 0c.225 0 .365.148.365.385v.799c0 .236-.14.384-.365.384h-.76c-.225 0-.393-.148-.365-.384v-.8c0-.236.14-.384.365-.384zm-9-3.136c.225 0 .365.148.365.385v.799c0 .236-.14.384-.365.384h-.76c-.225 0-.393-.177-.365-.384v-.8c0-.236.14-.384.365-.384zm2.981 0c.225 0 .366.148.366.385v.799c0 .236-.141.384-.366.384h-.76c-.224 0-.365-.177-.365-.384v-.8c0-.236.14-.384.366-.384zm3.01 0c.224 0 .365.148.365.385v.799c0 .236-.14.384-.366.384h-.759c-.225 0-.366-.177-.366-.384v-.8c0-.236.141-.384.366-.384zm3.009 0c.225 0 .365.148.365.385v.799c0 .236-.14.384-.365.384h-.76c-.225 0-.393-.177-.365-.384v-.8c0-.236.14-.384.365-.384zm2.587-5.532H1.491v2.366h14.99V1.598z"
                  fill="#242C39"
                />
              </svg>
            </div>
          </li>
        </a>
        <a
          href="/vehicles"
        >
          <li
            class="beat-ui-sidenav-item css-xqhjbi e1hw2vfw3"
            data-testid="nav-list-item"
          >
            <div
              class="css-16oynd e1hw2vfw5"
            >
              <svg
                height="14"
                viewBox="0 0 19 14"
                width="19"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="none"
                  fill-rule="evenodd"
                >
                  <g
                    fill="#242C39"
                    fill-rule="nonzero"
                  >
                    <g>
                      <path
                        d="M26.619 179l.366.005 6.005.022c.481-.025.964.012 1.435.112.728.204 1.325.723 1.625 1.413l1.244 3.244c.03.085.088.159.164.21.277.187.963.66 1.11.87.182.267.31.567.38.882.022.117.043 2.008.047 3.34v.347c0 .247.018.647.018.894 0 .456-.04.88-.349 1.25-.326.366-.804.562-1.296.53-.528 0-1.012-.01-1.428-.39-.417-.38-.48-.871-.48-1.399 0-.247-.201-.447-.45-.447h-9.999c-.248 0-.45.2-.45.447-.004.53-.063 1.01-.474 1.395-.365.342-.817.4-1.283.4-.466 0-.898-.031-1.276-.353-.353-.297-.482-.68-.52-1.102l-.008-.147.003-.191c.02-1.653.074-4.582.19-5.033l.013-.042c.168-.357.415-.672.723-.92.178-.124.364-.235.558-.334.095-.05.168-.135.205-.237l.51-1.363.329-.87c.204-.55.373-1.133.756-1.594.405-.483.975-.802 1.602-.894.364-.04.73-.053 1.096-.04zm10.296 6.05c-.143-.086-.319-.094-.47-.023l-1.685.796c-.079.037-.142.092-.19.156h-3.769l-.8.84-.8-.84h-3.803c-.047-.064-.11-.119-.19-.156l-1.685-.796-.093-.033c-.126-.032-.262-.013-.376.056-.255.152-.432.367-.5.637l-.042.176-.014.077c-.005.045-.005.084-.002.142.007.11.034.206.083.32.09.21.254.376.476.507.077.046.165.07.255.07h1.434l.085-.007c.084-.015.163-.05.23-.105l.251-.204.035-.034H28.8l1.2 1.238 1.2-1.238h3.422l.036.034.251.204c.09.073.2.112.315.112h1.434c.09 0 .178-.024.255-.07.222-.131.386-.298.476-.508.049-.113.076-.209.083-.32.005-.085.003-.13-.016-.218l-.042-.176c-.068-.27-.245-.485-.5-.637zm-3.092-4.55h-7.686c-.44 0-.828.287-.957.708l-.503 1.646c-.014.047-.021.096-.021.146 0 .276.223.5.5.5h9.648c.05 0 .1-.007.147-.022.264-.08.412-.36.332-.624l-.503-1.646c-.129-.42-.517-.708-.957-.708z"
                        transform="translate(-21 -179)"
                      />
                    </g>
                  </g>
                </g>
              </svg>
            </div>
          </li>
        </a>
        <a
          href="/geofence/1"
        >
          <li
            class="beat-ui-sidenav-item css-xqhjbi e1hw2vfw3"
            data-testid="nav-list-item"
          >
            <div
              class="css-16oynd e1hw2vfw5"
            >
              <svg
                height="16"
                viewBox="0 0 20 16"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18.847 3.91l-4.324-2.85c-.12-.08-.278-.08-.398 0L10 3.777 5.875 1.06c-.12-.079-.278-.079-.398 0L1.153 3.91c-.096.063-.153.168-.153.28v10.472c0 .126.072.241.188.3.116.058.255.05.363-.02l4.125-2.72 4.125 2.72c.12.079.278.079.398 0l4.125-2.72 4.125 2.72c.06.04.13.059.199.059.056 0 .112-.013.164-.04.116-.058.188-.173.188-.299V4.189c0-.112-.057-.217-.153-.28zm-.552 6.597l-1.549-3.065 1.55-.158v3.223h0zm0-3.563l-3.707.377V1.924l3.707 2.444v2.576h0zm-16.59-.2h3.707v3.221l-3.707-.493V6.744zm8.03-2.318v6.127L6.839 7.536 8.804 3.81l.932.615zm2.154-1.071l2.17.908v3.25l-3.795 3.125V4.426l1.625-1.071zm2.17.538l-1.827-.764 1.828-1.205v1.97h0zM6.585 7.27L5.94 6.6V1.924l2.57 1.694L6.585 7.27zM5.94 7.1l3.796 3.953v3.023L5.94 11.574V7.1zm4.324 3.984l3.796-3.125v3.615l-3.796 2.502v-2.992zm-4.852-9.16v4.481H1.705V4.368l3.707-2.444zm-3.707 7.89l3.707.494v1.266l-3.707 2.444V9.814zm12.883 1.76V7.662l1.786-.182 1.921 3.803v2.735l-3.707-2.444z"
                  fill="#242C39"
                  stroke="#242C39"
                  stroke-width=".6"
                />
              </svg>
            </div>
          </li>
        </a>
        <a
          href="/settings"
        >
          <li
            class="beat-ui-sidenav-item css-xqhjbi e1hw2vfw3"
            data-testid="nav-list-item"
          >
            <div
              class="css-16oynd e1hw2vfw5"
            >
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10.663 1c.669 0 1.212.544 1.212 1.212h0v1.345c0 .205.12.391.29.452.188.068.371.144.549.228.158.076.368.03.51-.112h0l.957-.958c.473-.471 1.241-.471 1.715 0h0l.937.938c.473.472.473 1.241 0 1.714h0l-.958.958c-.14.141-.186.35-.111.51.084.178.16.36.227.547.061.171.248.29.453.29h1.345c.668 0 1.212.544 1.211 1.213h0v1.326c0 .669-.544 1.212-1.212 1.212h0-1.345c-.205 0-.391.12-.453.29-.068.188-.144.37-.228.548-.075.16-.03.369.112.51h0l.958.958c.473.473.472 1.242 0 1.714h0l-.937.938c-.473.47-1.241.472-1.715 0h0l-.957-.958c-.14-.141-.35-.188-.51-.112-.178.085-.36.16-.547.227-.172.063-.291.249-.291.453h0v1.345c0 .668-.544 1.212-1.212 1.212h0-1.326c-.669 0-1.212-.543-1.212-1.212h0v-1.345c0-.205-.12-.391-.29-.453-.188-.068-.371-.144-.549-.228-.159-.075-.368-.029-.51.112h0l-.957.958c-.473.47-1.241.472-1.715 0h0l-.937-.938c-.472-.472-.472-1.24 0-1.714h0l.958-.958c.141-.14.187-.35.112-.51-.084-.177-.16-.36-.227-.546-.062-.172-.248-.291-.453-.291h0-1.345C1.544 11.875 1 11.33 1 10.663h0V9.337c0-.669.544-1.212 1.212-1.212h1.345c.205 0 .391-.12.452-.29.068-.188.144-.37.228-.548.075-.16.03-.369-.11-.51h0l-.96-.958c-.472-.473-.472-1.242 0-1.714h0l.937-.938c.473-.471 1.242-.471 1.715 0h0l.958.958c.14.141.35.187.51.112.178-.085.36-.16.547-.227.172-.063.291-.249.291-.453h0V2.212C8.125 1.544 8.67 1 9.337 1h0zm0 .75H9.337c-.255 0-.462.207-.462.462h0v1.345c0 .523-.315.99-.786 1.158-.164.06-.324.126-.481.2-.448.212-.994.106-1.36-.26h0l-.959-.957c-.18-.18-.474-.18-.654 0h0l-.937.937c-.18.18-.18.473 0 .653h0l.958.958c.366.365.47.911.26 1.36-.075.157-.141.318-.201.483-.169.47-.635.786-1.158.786h0-1.345c-.255 0-.462.207-.462.462h0v1.326c0 .255.207.462.462.462h1.345c.523 0 .99.315 1.159.786.059.165.126.325.2.483.21.447.106.994-.26 1.36h0l-.958.956c-.18.18-.18.474 0 .654h0l.937.938c.182.18.475.18.654 0h0l.958-.958c.241-.24.558-.368.875-.368.166 0 .331.036.485.108.158.075.318.141.483.201.47.17.785.635.785 1.158h0v1.345c0 .255.207.462.462.462h1.326c.255 0 .462-.207.462-.462h0v-1.345c0-.523.316-.989.786-1.158.165-.06.325-.126.482-.2.448-.21.995-.106 1.36.26h0l.958.957c.182.18.475.18.654 0h0l.938-.937c.18-.18.18-.474 0-.654h0l-.958-.958c-.366-.365-.47-.911-.26-1.36.075-.157.142-.317.2-.482.17-.47.635-.786 1.159-.786h1.345c.255 0 .462-.207.462-.462h0V9.337c0-.255-.207-.462-.462-.462h0-1.345c-.524 0-.99-.316-1.159-.787-.06-.164-.126-.325-.2-.482-.21-.448-.106-.995.26-1.36h0l.958-.958c.18-.18.18-.473 0-.653h0l-.938-.937c-.18-.18-.474-.18-.654 0h0l-.957.957c-.366.367-.913.472-1.36.26-.158-.074-.318-.14-.483-.2-.47-.17-.786-.635-.786-1.158h0V2.212c0-.255-.207-.462-.462-.462h0zM10 7.375c1.448 0 2.625 1.177 2.625 2.625 0 1.447-1.177 2.625-2.625 2.625-1.447 0-2.625-1.178-2.625-2.625 0-1.448 1.178-2.625 2.625-2.625zm0 .75c-1.033 0-1.875.84-1.875 1.875 0 1.034.842 1.875 1.875 1.875 1.034 0 1.875-.841 1.875-1.875 0-1.035-.841-1.875-1.875-1.875z"
                  fill="#242C39"
                  stroke="#242C39"
                  stroke-width=".6"
                />
              </svg>
            </div>
          </li>
        </a>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
